File: Tugas 12/ApiClient/AppPerpustakaan/Peminjaman.py
import requests
import json
class Peminjaman:
    def __init__(self):
        self.__id=None
        self.__NIM = None
        self.__Tanggal_pinjam = None
        self.__Tanggal_kembali = None
        self.__Id_buku = None
        self.__Id_petugas = None
        self.__Id_anggota = None
        self.__url = "http://localhost/Perpustakaan/peminjaman_api.php"
                    
    @property
    def id(self):
        return self.__id
    @property
    def NIM(self):
        return self.__NIM
        
    @NIM.setter
    def NIM(self, value):
        self.__NIM = value
    @property
    def Tanggal_pinjam(self):
        return self.__Tanggal_pinjam
        
    @Tanggal_pinjam.setter
    def Tanggal_pinjam(self, value):
        self.__Tanggal_pinjam = value
    @property
    def Tanggal_kembali(self):
        return self.__Tanggal_kembali
        
    @Tanggal_kembali.setter
    def Tanggal_kembali(self, value):
        self.__Tanggal_kembali = value
    @property
    def Id_buku(self):
        return self.__Id_buku
        
    @Id_buku.setter
    def Id_buku(self, value):
        self.__Id_buku = value
    @property
    def Id_petugas(self):
        return self.__Id_petugas
        
    @Id_petugas.setter
    def Id_petugas(self, value):
        self.__Id_petugas = value
    @property
    def Id_anggota(self):
        return self.__Id_anggota
        
    @Id_anggota.setter
    def Id_anggota(self, value):
        self.__Id_anggota = value
    def get_all(self):
        payload ={}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(self.__url, json=payload, headers=headers)
        return response.text
    def get_by_NIM(self, NIM):
        url = self.__url+"?NIM="+NIM
        payload = {}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(url, json=payload, headers=headers)
        data = json.loads(response.text)
        for item in data:
            self.__id = item['Id_peminjaman']
            self.__NIM = item['NIM']
            self.__Tanggal_pinjam = item['Tanggal_pinjam']
            self.__Tanggal_kembali = item['Tanggal_kembali']
            self.__Id_buku = item['Id_buku']
            self.__Id_petugas = item['Id_petugas']
            self.__Id_anggota = item['Id_anggota']
        return data
    def simpan(self):
        payload = {
            "NIM":self.__NIM,
            "Tanggal_pinjam":self.__Tanggal_pinjam,
            "Tanggal_kembali":self.__Tanggal_kembali,
            "Id_buku":self.__Id_buku,
            "Id_petugas":self.__Id_petugas,
            "Id_anggota":self.__Id_anggota
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.post(self.__url, data=payload, headers=headers)
        return response.text
    def update_by_NIM(self, NIM):
        url = self.__url+"?NIM="+NIM
        payload = {
            "NIM":self.__NIM,
            "Tanggal_pinjam":self.__Tanggal_pinjam,
            "Tanggal_kembali":self.__Tanggal_kembali,
            "Id_buku":self.__Id_buku,
            "Id_petugas":self.__Id_petugas,
            "Id_anggota":self.__Id_anggota
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.put(url, data=payload, headers=headers)
        return response.text
    def delete_by_NIM(self,NIM):
        url = self.__url+"?NIM="+NIM
        headers = {'Content-Type': 'application/json'}
        payload={}
        response = requests.delete(url, json=payload, headers=headers)
        return response.text

File: Tugas Aplikasi Perpustakaan/Peminjaman.php
<?php
//Simpanlah dengan nama file : Peminjaman.php
require_once 'database.php';
class Peminjaman 
{
    private $db;
    private $table = 'peminjaman';
    public $NIM = "";
    public $Tanggal_pinjam = "";
    public $Tanggal_kembali = "";
    public $Id_buku = "";
    public $Id_petugas = "";
    public $Id_anggota = "";
    public function __construct(MySQLDatabase $db)
    {
        $this->db = $db;
    }
    public function get_all() 
    {
        $query = "SELECT * FROM $this->table";
        $result_set = $this->db->query($query);
        return $result_set;
    }
    public function get_by_id(int $id)
    {
        $query = "SELECT * FROM $this->table WHERE id = $id";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function get_by_NIM(int $NIM)
    {
        $query = "SELECT * FROM $this->table WHERE NIM = $NIM";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function insert(): int
    {
        $query = "INSERT INTO $this->table (`NIM`,`Tanggal_pinjam`,`Tanggal_kembali`,`Id_buku`,`Id_petugas`,`Id_anggota`) VALUES ('$this->NIM','$this->Tanggal_pinjam','$this->Tanggal_kembali','$this->Id_buku','$this->Id_petugas','$this->Id_anggota')";
        $this->db->query($query);
        return $this->db->insert_id();
    }
    public function update(int $id): int
    {
        $query = "UPDATE $this->table SET NIM = '$this->NIM', Tanggal_pinjam = '$this->Tanggal_pinjam', Tanggal_kembali = '$this->Tanggal_kembali', Id_buku = '$this->Id_buku', Id_petugas = '$this->Id_petugas', Id_anggota = '$this->Id_anggota' 
        WHERE Id_peminjaman = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function update_by_NIM($NIM): int
    {
        $query = "UPDATE $this->table SET NIM = '$this->NIM', Tanggal_pinjam = '$this->Tanggal_pinjam', Tanggal_kembali = '$this->Tanggal_kembali', Id_buku = '$this->Id_buku', Id_petugas = '$this->Id_petugas', Id_anggota = '$this->Id_anggota' 
        WHERE NIM = $NIM";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete(int $id): int
    {
        $query = "DELETE FROM $this->table WHERE Id_peminjaman = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete_by_NIM($NIM): int
    {
        $query = "DELETE FROM $this->table WHERE NIM = $NIM";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
}
?>